torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -976,6 +981,35 @@ def test_profiler_with_script_remote_rpc(self):
             use_record_function=True,
         )
 
+    @dist_init
+    def test_server_process_global_profiler(self):
+        if self.rank != 0:
+            return
+
+        dst_rank = (self.rank + 1) % self.world_size
+        dst_worker_name = worker_name(dst_rank)
+
+        x = torch.tensor(1)
+        y = torch.tensor(2)
+
+        rpc.rpc_sync(dst_worker_name, enable_profiler)
+        rpc.rpc_sync(dst_worker_name, torch.add, (x, y))
+        events = rpc.rpc_sync(dst_worker_name, rpc.disable_server_process_global_profiler)
```

Comment:
What happens in the case where multiple workers send RPCs to the server at the same time? Are we guaranteed that only profiling results returned to a worker will be due to work they initiated on the server? Will this invoke the overhead of profiling other workers' requests even if they have not specifically requested for that?

Reply:
@rohan-varma 
No matter which client worker or whether the client enables profiling or not, all threads that handle requests do profiling.
That's why I am calling it server global profiling.

This overhead is ok for our use case. When we enable this on an embedding server, all the RPCs the server do is embedding lookup forward and backward passes. If there is more types of RPCs involved like signaling/metric reporting, we can optimize those later.